### PR TITLE
Update `Path`, `PathList`, and `IsInstance` Domain Validators

### PR DESCRIPTION
**Pull request by shermanjasonaf**
## Summary/Motivation:

This PR is a follow-up to #3140, itself motivated by discussion in #3126.

## Changes proposed in this PR:
- Add `IsInstance` to the `config` library reference in the online documentation.
- Make the usage of fully qualified valid type names in `IsInstance.domain_name()` and any error messages optional (as sometimes, the fully qualified names can be very long).
- Implement `Path.domain_name()`.
- Make documentation of `Path` more consistent with updates of #3140.
- Adapt behavior and documentation of `PathList` to updated implementation of `Path`.

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by mrmundt — APPROVED** [review]

File: pyomo/common/config.py [hunk 1]
```diff
@@ -562,10 +589,15 @@ class PathList(Path):
...
-            return [super(PathList, self).__call__(data)]
+        try:
+            pathlist = [super(PathList, self).__call__(data)]
+        except TypeError as err:
+            is_not_path_like = "expected str, bytes or os.PathLike" in str(err)
+            if is_not_path_like and hasattr(data, "__iter__"):
+                pathlist = [super(PathList, self).__call__(i) for i in data]
+            else:
+                raise
+        return pathlist
```
Comment (jsiirola) [inline pyomo/common/config.py:600]: What is the motivation for this change (especially because I don't see [an obvious] new test that would pass here and fail before)?  I am concerned about putting in a try-except in the validator that is expected to fail in normal cases.

Comment (shermanjasonaf) [reply pyomo/common/config.py:600]: In lieu of the changes, I am thinking about changing `PathList.__call__` to something like:

```python
def __call__(self, data):
    is_path_like = isinstance(data, (str, bytes)) or hasattr(data, "__fspath__")
    if is_path_like:
        return [super(PathList, self).__call__(data)]
    else:
        return [super(PathList, self).__call__(i) for i in data]
```

Thoughts?

Comment (jsiirola) [reply pyomo/common/config.py:600]: (we were both concurrently  thinking along the exact same lines)

**Review by jsiirola — APPROVED** [review]
